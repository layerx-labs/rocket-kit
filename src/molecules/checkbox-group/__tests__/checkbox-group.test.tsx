import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { CheckboxGroup } from '../../..';

const checkboxButtons = [
  { label: 'Option 1', value: 'option_1', checked: false, disabled: false },
  { label: 'Option 2', value: 'option_2', checked: false, disabled: false },
];

describe('CheckboxGroup', () => {
  it('renders', () => {
    const { asFragment } = render(<CheckboxGroup options={checkboxButtons} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
