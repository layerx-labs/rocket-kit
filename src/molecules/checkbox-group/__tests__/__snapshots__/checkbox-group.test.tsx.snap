// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup renders 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  padding-left: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1:hover input:not(:disabled) ~ span {
  border-color: #000000;
}

.c2 {
  margin-left: calc(1.5rem + 0.3125rem);
  line-height: 1.5rem;
}

.c3 {
  position: absolute;
  left: 0;
  opacity: 0;
  cursor: pointer;
}

.c3:checked:not(:disabled) ~ span {
  background-color: #56cad7;
  border-color: #2ba8b6;
}

.c3:checked:not(:disabled) ~ span:after {
  display: block;
}

.c3:checked:disabled ~ span {
  border-color: transparent;
  background-color: #c4c7d3;
  color: #939393;
}

.c3:checked:disabled ~ span:after {
  display: block;
}

.c3:not(:checked):disabled ~ span {
  border-color: #c4c7d3;
  background-color: #ffffff;
}

.c3:hover:not(:disabled) {
  border-color: #2ba8b6;
}

.c3:hover:not(:disabled):checked ~ span {
  border-color: #2ba8b6;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  border: 0.125rem solid #939393;
  border-radius: 100%;
  width: 1.5rem;
  height: 1.5rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c4:after {
  content: '';
  position: absolute;
  display: none;
  top: 0.1875rem;
  left: 0.375rem;
  width: 0.3125rem;
  height: 0.5625rem;
  border: solid #ffffff;
  border-width: 0 0.1875rem 0.1875rem 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 li {
  position: relative;
  min-height: 1.5rem;
}

.c0 li:not(:last-child) {
  margin-bottom: 0.625rem;
}

<ul
    class="c0"
    type="column"
  >
    <li>
      <label
        class="c1 checkbox"
      >
        <span
          class="c2"
        >
          Option 1
        </span>
        <input
          aria-checked="false"
          class="c3"
          name="option_1"
          type="checkbox"
        />
        <span
          class="c4"
        />
      </label>
    </li>
    <li>
      <label
        class="c1 checkbox"
      >
        <span
          class="c2"
        >
          Option 2
        </span>
        <input
          aria-checked="false"
          class="c3"
          name="option_2"
          type="checkbox"
        />
        <span
          class="c4"
        />
      </label>
    </li>
  </ul>
</DocumentFragment>
`;
